packages/core/src/plugins/meta/discogs.test.ts

```diff
@@ -0,0 +1,164 @@
+import DiscogsMetaProvider from './discogs';
+import { SimilarArtist } from '../plugins.types';
+import * as SpotifyApi from '../../rest/Spotify';
+
+describe('Tests for DiscogsMetaProvider', () => {
+  const provider = new DiscogsMetaProvider();
+
+  describe('Verify if an artist is on tour', () => {
+
+    test('Should return false if the artist info is missing', () => {
+      expect(provider.isArtistOnTour(undefined)).toBeFalsy();
+    });
+
+    test('Should return false if the \'ontour\' flag is not \'1\'', () => {
+      const artistInfo: any = {
+        ontour: 'something else than 1'
+      };
+      expect(provider.isArtistOnTour(artistInfo)).toBeFalsy();
+    });
+
+    test('Should return false if the \'ontour\' flag is \'1\'', () => {
+      const artistInfo: any = {
+        ontour: '1'
+      };
+      expect(provider.isArtistOnTour(artistInfo)).toBeTruthy();
+    });
+  });
+
+  describe('Get similar artists', () => {
+    const spotifyToken = 'spotify token';
+    const getToken = jest.spyOn(SpotifyApi, 'getToken');
+    const fetchTopSimilarArtistsFromSpotify = jest.spyOn(provider, 'fetchTopSimilarArtistsFromSpotify');
+
+    afterEach(() => {
+      getToken.mockReset();
+      fetchTopSimilarArtistsFromSpotify.mockReset();
+    });
+
+    afterAll(() => {
+      getToken.mockRestore();
+      fetchTopSimilarArtistsFromSpotify.mockRestore();
+    });
+
+    test('Should return an empty result set if the artist info is missing', async () => {
+      expect(await provider.getSimilarArtists({
+        similar: {
+          artist: null
+        }
+      } as any)).toEqual([]);
```

Comment:
It added it to allow passing the object as an argument and to avoid defining unnecessary properties. I'll change it to ` as LastFmArtistInfo` . Thanks for pointing it out.